static/cfg-view.js

```diff
@@ -0,0 +1,217 @@
+                                             this.defaultCfgOuput, this.networkOpts);
+        
+        this.restButton = this.domRoot.find(".show-hide-btn")
+                .on('click', _.bind(function(event){
+                    this.networkOpts.interaction.navigationButtons = !this.networkOpts.interaction.navigationButtons;
+                    this.cfgVisualiser.setOptions(this.networkOpts);
+                }, this));
+        
+        this._compilerid = state.id;
+        this._compilerName = state.compilerName;
+        this._editorid = state.editorid;
+        
+        this.eventHub.on('compileResult', this.onCompileResult, this);
+        this.eventHub.on('compiler', this.onCompiler, this);
+        this.eventHub.on('compilerClose', this.onCompilerClose, this);
+        //this.eventHub.on('editorChange', this.onEditorChange, this);
+        this.eventHub.emit('cfgViewOpened', this._compilerid);
+        this.container.on('destroy', function () {
+            this.eventHub.emit("cfgViewClosed", this._compilerid, this.cfgVisualiser);
+            this.eventHub.unsubscribe();
+        }, this);
+        
+        this.adaptStructure = function(names){
+            var options = [];
+            
+            for(var i = 0; i < names.length; ++i){
+                options.push({name:names[i]});
+            }
+            return options;
+        };
+        
+        
+        
+        var self = this;
+        
+        this.select = this.domRoot.find(".function-picker").selectize({
+            sortField: 'name',
+            valueField: 'name',
+            labelField: 'name',
+            searchField: ['name'],
+            options: this.fnNames.length ? this.adaptStructure(this.fnNames) : [{name:"no functions"}],
+            items: this.fnNames.length ? [this.currentFunc] : [{name:"no functions"}]
+        }).on('change', function(event){
+            self.onFunctionChange(self.functions, this.value);
+        });
+        
+        this.setTitle();
+   }
+   
+   Cfg.prototype.onCompileResult = function (id, compiler, result) {
+        if (this._compilerid === id) {
+            if (result.supportCfg && !$.isEmptyObject(result.cfg)) {
+                this.functions = result.cfg;
+                this.fnNames = Object.keys(this.functions);
+                if(!this.fnNames.includes(this.currentFunc))
+                    this.currentFunc = this.fnNames[0];
+                this.showCfgResults({
+                    'nodes': this.functions[this.currentFunc].nodes,
+                    'edges': this.functions[this.currentFunc].edges
+                });
+                
+
+                
+            } else {
+                this.showCfgResults(this.defaultCfgOuput);
+                this.currentFunc = "";
+                this.fnNames = [];
+            }
+            
+            this.domRoot.find(".function-picker")[0].selectize.destroy();
```

Comment:
if you have repeated calls to find best to cache the result.

Reply:
done